conda_lock/conda_solver.py

```diff
@@ -424,7 +424,7 @@ def update_specs_for_arch(
             entry["name"]: entry
             for entry in json.loads(
                 subprocess.check_output(
-                    [str(conda), "list", "-p", prefix, "--json"],
+                    [str(conda), "list", "--no-pip", "-p", prefix, "--json"],
```

Comment:
Note that `--no-pip` is new in mamba [v2.0.6](https://github.com/mamba-org/mamba/releases/tag/2.0.6). We need to figure out how to handle the cases `mamba >=2,<2.0.6` and `mamba <2`.